cluster-autoscaler/core/scaledown/legacy/legacy.go

```diff
@@ -282,7 +282,10 @@ func (sd *ScaleDown) NodesToDelete(currentTime time.Time, pdbs []*policyv1.PodDi
 	}
 
 	scaleDownResourcesLeft := sd.resourceLimitsFinder.LimitsLeft(sd.context, allNodes, resourceLimiter, currentTime)
-	empty, nonEmpty, unremovable := sd.unneededNodes.RemovableAt(sd.context, currentTime, scaleDownResourcesLeft, resourceLimiter.GetResources(), sd.nodeDeletionTracker)
+	emptyToRemove, nonEmpty, unremovable := sd.unneededNodes.RemovableAt(sd.context, currentTime, scaleDownResourcesLeft, resourceLimiter.GetResources(), sd.nodeDeletionTracker)
+	for _, emptyNode := range emptyToRemove {
```

Comment:
Do we need the `empty` list? Isn't it only used to extract the name in the loop below? Couldn't we just do the same thing as with `nonEmpty`?

Reply:
Actually it is also used as a return type, so we do need that.

Reply:
I mean we're setting the variable here, but `empty` isn't returned anywhere from this function, right? It's only used to create `candidateNames`. Or am I reading things wrong?

Reply:
discussed offline, removed empty